src/core/iTextSharp/text/pdf/security/X509Certificate2Signature.cs

```diff
@@ -97,20 +97,18 @@ public virtual byte[] Sign(byte[] message)
                 // Modified by J. Arturo
                 // Workaround for SHA-256 and SHA-512
 
-                if (rsa.CspKeyContainerInfo.ProviderName == "Microsoft Strong Cryptographic Provider" ||
-                                rsa.CspKeyContainerInfo.ProviderName == "Microsoft Enhanced Cryptographic Provider v1.0" ||
-                                rsa.CspKeyContainerInfo.ProviderName == "Microsoft Base Cryptographic Provider v1.0")
+                if (rsa.CspKeyContainerInfo.ProviderName == CryptoConst.MS_STRONG_PROV ||
+                     rsa.CspKeyContainerInfo.ProviderName == CryptoConst.MS_ENHANCED_PROV ||
+                     rsa.CspKeyContainerInfo.ProviderName == CryptoConst.MS_DEF_PROV)
```

Comment:
I don't know if this could be replaced by:

`if (!rsa.CspKeyContainerInfo.HardwareDevice) {`